packages/remark-mdx/tag.js

```diff
@@ -7,7 +7,7 @@ const attributeName = '[a-zA-Z_:][a-zA-Z0-9:._-]*'
 const unquoted = '[^"\'=<>`\\u0000-\\u0020]+'
 const singleQuoted = "'[^']*'"
 const doubleQuoted = '"[^"]*"'
-const objectProps = '{{[^}]*}}'
+const jsProps = '{[^\n]*}'
```

Comment:
Accept anything inside `props={/* ... */}` to handle everything from functions, to string interpolation, to nested objects.